contracts/eosio.msig/src/eosio.msig.cpp

```diff
@@ -16,12 +23,17 @@ void multisig::propose( ignore<name> proposer,
    std::vector<permission_level> _requested;
 
    _ds >> _proposer >> _proposal_name >> _requested;
+   require_auth( _proposer );
 
    const char* trx_pos = _ds.pos();
    size_t size = _ds.remaining();
-   transaction_header _trx_header = get_trx_header(trx_pos, size);
-   require_auth( _proposer );
-   check( _trx_header.expiration >= eosio::time_point_sec(current_time_point()), "transaction expired" );
+
+   transaction_header trx_header;
+   std::vector<action> context_free_actions;
+   _ds >> trx_header;
+   check( trx_header.expiration >= eosio::time_point_sec(current_time_point()), "transaction expired" );
+   _ds >> context_free_actions;
```

Comment:
It would be cheaper to just deserialize an `eosio::unsigned_int` and check that its value is 0 rather than deserializing all of the context-free actions. On the other hand, it would only be expensive to deserialize if the `check` fails and therefore the transaction does not retire on the blockchain (outside of the context of deferred transactions anyway). So you actually don't give up performance in practice by just deserializing the entire `context_free_actions` and it makes it clearer what the code is doing. So I will leave it up to you to decide if you want to make any change here.

Reply:
I opted to leave it as is.